pkg/listwatch/listwatch.go

```diff
@@ -66,15 +62,22 @@ func NewUnprivilegedNamespaceListWatchFromClient(l log.Logger, c cache.Getter, a
 func NewFilteredUnprivilegedNamespaceListWatchFromClient(l log.Logger, c cache.Getter, allowedNamespaces, deniedNamespaces map[string]struct{}, optionsModifier func(options *metav1.ListOptions)) cache.ListerWatcher {
 	// If the only namespace given is `v1.NamespaceAll`, then this
 	// cache.ListWatch must be privileged. In this case, return a regular
-	// cache.ListWatch decorated with a denylist watcher
+	// cache.ListWatch tweaked with denylist fieldselector
 	// filtering the given denied namespaces.
 	if IsAllNamespaces(allowedNamespaces) {
-		return newDenylistListerWatcher(
-			l,
-			deniedNamespaces,
-			cache.NewFilteredListWatchFromClient(c, "namespaces", metav1.NamespaceAll, optionsModifier),
-		)
+		tweak := func(options *metav1.ListOptions) {
+			if optionsModifier != nil {
+				optionsModifier(options)
+			}
+
+			fieldSelector := options.FieldSelector
+			denyNamesTweak(options, deniedNamespaces)
+			options.FieldSelector = strings.Join([]string{options.FieldSelector, fieldSelector}, ",")
```

Comment:
couldn't this be done in `denyNamesTweak()`?

Reply:
indeed, I unified the methods